### ISSUE TITLE
i don't think coupling db with bot client is a good idea

### ISSUE THREAD

**Issue by ningskyer** [php-telegram-bot/core#817]
as saying up

**Comment by akalongman** [comment]
Any suggestion? We are thinking make storage abstract, but its not so easy

**Comment by skybebe** [comment]
u could even think of a way to implement the bot client without a db or storage。I don't need any storage In my business any way.

**Comment by akalongman** [comment]
@skybebe you maybe not, but there are many people who asked for adding some storage layer

**Comment by jacklul** [comment]
I think it should be an optional plugin anyway, I eventually moved away from using it and implement own database handler in my projects...

**Comment by skybebe** [comment]
yes, that's exactly what i mean.  bot is bot , db is db, whether i need db or not. if i need db, i can configure it.
Maybe this is a better way
